Mathlib/MeasureTheory/Group/FundamentalDomain.lean

```diff
@@ -719,4 +727,315 @@ protected theorem fundamentalInterior : IsFundamentalDomain G (fundamentalInteri
 
 end IsFundamentalDomain
 
+section FundamentalDomainMeasure
+
+variable (G) [Group G] [MulAction G α] [MeasurableSpace α] {s : Set α}
+  (μ : Measure α) (hs : NullMeasurableSet s μ)
+
+local notation "α_mod_G" => MulAction.orbitRel G α
+
+local notation "π" => @Quotient.mk _ α_mod_G
+
+/-- Given a measurable subset `s` of a `MeasurableSpace` `α`, a group `G` acting on `α`, and a
+measure `μ` on `α`, one can define a `Measure` on the `Quotient` of `α ⧸ G` by lifting to `α` and
+intersecting with `s`. -/
+@[to_additive NullMeasurableSet.addQuotientMeasure "Given a measurable subset `s` of a
+`MeasurableSpace` `α`, an additive group `G` acting on `α`, and a measure `μ` on `α`, one can define
+a `Measure` on the `Quotient` of `α ⧸ G` by lifting to `α` and intersecting with `s`."]
+noncomputable def NullMeasurableSet.quotientMeasure : Measure (Quotient α_mod_G) := by
+  apply Measure.ofMeasurable (fun U _ => μ ((π ⁻¹' U) ∩ s)) (by simp)
+  intro f meas_f disjoint_f
+  let ff : ℕ → Set α := fun n => π ⁻¹' (f n) ∩ s
+  have meas_ff : ∀ n, NullMeasurableSet (ff n) μ :=
+    fun n => (measurableSet_quotient.mp (meas_f n)).nullMeasurableSet.inter hs
+  have disjoint_ff : Pairwise (AEDisjoint μ on fun (n : ℕ) => ff n)
+  · intro x y x_ne_y
+    have := (Disjoint.preimage π (disjoint_f x_ne_y))
+    rw [Function.onFun_apply]
+    have : Disjoint (ff x) (ff y) := by dsimp [Disjoint]; aesop
+    exact this.aedisjoint
+  convert measure_iUnion₀ disjoint_ff meas_ff using 1
+  simp [iUnion_inter]
```

Comment:
Is this the same as `(μ.map π).restrict s`?

Reply:
Good point, we'll update the file accordingly.

Reply:
Ok, file is updated. How does it look to you?